modules/queue/queue_redis.go

```diff
@@ -96,12 +99,12 @@ func (fifo *RedisByteFIFO) PushFunc(data []byte, fn func() error) error {
 			return err
 		}
 	}
-	return fifo.client.RPush(fifo.queueName, data).Err()
+	return fifo.client.RPush(graceful.GetManager().HammerContext(), fifo.queueName, data).Err()
```

Comment:
This one - now here you need to change the ByteFIFOQueue to have a function ShutdownContext() that returns a context.Context wrapping the q.closed channel and is Done() when the q.closed channel is closed.

That should be the context passed here.

Reply:
There's a ChannelContext in modules/graceful that can do this.